src/sentry/incidents/logic.py

```diff
@@ -843,6 +841,9 @@ def update_alert_rule(
                 project for project in projects if project.slug not in existing_project_slugs
             ]
             updated_project_slugs = {project.slug for project in projects}
+            alert_rule.projects.filter(project__slug__not__in=updated_project_slugs).delete()
+            for project in projects:
+                alert_rule.projects.add(project)
```

Comment:
Have we updated everywhere that's looking at alert rule projects via snuba?

if not or not sure, we may not want to start deleting the projects filter yet. I updated some places to read the project from alert rules, but I only checked the 1 API flow.

Reply:
Not yet!
This PR is just to ensure that all alert rules will now be assured to _have_ projects

will need an update PR after the backfill migration to move everythign over to the AlertRule.projects rather than the subscription.project

Reply:
ah okay, so this will just delete it if we are going to migrate it over 👍